docs/autostart_win.rst

```diff
@@ -0,0 +1,105 @@
+Setting up auto-restart on Windows
+==================================
+
+-------------------------------
+Downloading NSSM via Chocolatey
+-------------------------------
+
+.. note:: This assumes you already have chocolatey installed. 
+          If not, you may install it or follow the manual instructions for getting NSSM.
+
+To install via Chocolatey, search "powershell" in the start menu,
+right-click on it and then click "Run as administrator"
+
+Then run the following command:
+
+.. code-block:: none
+
+    choco install nssm -y
+
+------------------------
+Manually installing NSSM
+------------------------
+
+Go to `<https://nssm.cc/download>`_ and download the latest release. Once the download
+is finished, open File Explorer, navigate to the drive Windows is installed on, and 
+create a folder called :code:`nssm`. Then navigate to your Downloads folder, and move 
+:code:`nssm.exe` to :code:`C:\nssm`.
+
+Then open Powershell as administrator (search "powershell" in the start menu, then right-click and 
+click "Run as administrator") and run the following commands to add NSSM to your path.
+
+.. code-block:: none
+
+    $oldPath = (Get-Itemproperty -path 'hklm:\system\currentcontrolset\control\session manager\environment' -Name Path).Path
+    $newPath = $oldPath + ";C:\nssm"
+    Set-ItemProperty -path 'hklm:\system\currentcontrolset\control\session manager\environment' -Name Path -Value $newPath
+
+Once finished, log out and back in to ensure the changes take effect.
+
+----------------------
+Setting up the service
+----------------------
+
+To create the service, run the following command:
+
+.. note:: If Red was installed in a venv, please activate it before running this command
+
+.. code-block:: none
+
+    nssm install redbot $(Get-Command redbot).Definition <instance name> --no-prompt
+
+replacing <instance name> with the name of the instance.
+
+Then run the following set of commands one at a time:
+
+.. code-block:: none
+
+    nssm set redbot AppExit 0 Exit
+    nssm set redbot AppStopMethodConsole 10000
+    nssm set redbot AppRestartDelay 15000
+    nssm set redbot AppStdout $Env:USERPROFILE\red-stdout.log
+    nssm set redbot AppStderr $Env:USERPROFILE\red-stderr.log
+    nssm set redbot AppNoConsole 1
+
+The above commands change a number of settings for the service including:
+
+- Ensuring that when ``[p]shutdown`` is run, the service doesn't attempt to restart
+
+- Ensuring that if Control-C is pressed for the process, nssm waits 10 seconds
+
+- Setting a 15 second delay for service restart
+
+- Setting a couple of log files for the service
+
+- Disabling the console window
+
+Now that the service has been configured, run the following command:
+
+.. code-block:: none
+
+    nssm set redbot ObjectName $Env:COMPUTERNAME\$Env:USERNAME <password>
+
+replacing <password> with the password you use to login to your computer. This
+sets the service to run under your user account so it can find your instances.
```

Comment:
I personally wouldn't want to put my password in cmd/powershell as part of a command (one of the reason being it saving in command history), I think there should at least be mentioned an alternative way to set that password through `services.msc`.

Also, both PowerShell and cmd have certain characters that have to be escaped, currently provided solution assumes that user's password won't contain such characters. I think that a better solution here would be using something that will prompt for password, e.g:
```
nssm set redbot ObjectName $Env:COMPUTERNAME\$Env:USERNAME $(Read-Host Password)
```

This isn't perfect - such prompt would still keep the password visible (as `input("Password: ")` does in Python), but it would at least always get the password right. The alternative that would show typed password with asterixes can be done in `Read-Host` by using `-AsSecureString` but it would result in this awfully long command, so the first solution might be good enough:
```
nssm set redbot ObjectName $Env:COMPUTERNAME\$Env:USERNAME $([System.Net.NetworkCredential]::new("", $(Read-Host -AsSecureString "Password")).Password)
```

---

One more issue that user may run into is having no password on the account - in such case, they won't be able to run service on an account with a blank password as it could be considered a security issue (see: <https://stackoverflow.com/a/1048691>)
While we probably shouldn't provide solution for it, it might be worth mentioning in the document.

Reply:
I'm thinking 
```pwsh
$c = Get-Credential
nssm set redbot ObjectName $c.Username $c.Password
```
might work here but I need to test it

Reply:
Same as the `Read-Host -AsSecureString`, `$c.Password` is also a secure string which sadly requires doing something like this:
```
$([System.Net.NetworkCredential]::new("", $c.Password)).Password
```
to get the actual raw string